lang/js/lib/protocols.js

```diff
@@ -1099,6 +1099,8 @@ MessageDecoder.prototype._flush = function () {
     this.emit('error', new Error('trailing data'));
   } else if (this._empty) {
     this.emit('error', new Error('no message decoded'));
+  } else {
+    this.emit('finish');
```

Comment:
This fixes the issue with Node.js 16 but it is a mystery to me why `finish` event is not emitted automatically on the Transform stream after the reabable.end().